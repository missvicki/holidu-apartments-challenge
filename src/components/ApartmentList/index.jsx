import React from 'react';
import ApartmentCard from '../ApartmentCard';

function ApartmentList(apartments){
    const apartmentList = apartments.apartments;
    if(apartmentList){
        return apartmentList.map(apartment => (
            <ApartmentCard 
                key={apartment.id} 
                name={apartment.details.name}
                location={apartment.location.name}
                price={apartment.price.total}
                currency={apartment.price.currency}
                image={apartment.photos[0].m}
                rating={apartment.rating.count}
                latitude={apartment.location.lat}
                longitude={apartment.location.lng}
                bedrooms={apartment.details.bedroomsCount}
                guests={apartment.details.guestsCount}
                title={apartment.details.apartmentTypeTitle}
                />
            ))
    }
}
export default ApartmentList;
