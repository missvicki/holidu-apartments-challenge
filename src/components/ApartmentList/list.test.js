import React from 'react';
import { shallow } from 'enzyme';
import ApartmentList from './index';

describe('List', () => {
  it('should not have regressed', () => {
    const component = shallow(<ApartmentList />);

    expect(component).toMatchSnapshot();
  });
});
