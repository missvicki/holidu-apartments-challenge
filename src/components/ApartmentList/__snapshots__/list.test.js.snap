// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should not have regressed 1`] = `""`;
